src/test/java/com/neo4j/docker/TestSSL.java

```diff
@@ -0,0 +1,165 @@
+package com.neo4j.docker;
+
+import com.neo4j.docker.utils.HostFileSystemOperations;
+import com.neo4j.docker.utils.Neo4jVersion;
+import com.neo4j.docker.utils.SetContainerUser;
+import com.neo4j.docker.utils.TestSettings;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+
+
+public class TestSSL
+{
+	private static final Logger log = LoggerFactory.getLogger( TestSSL.class );
+
+	@BeforeAll
+	static void filterOutPre40()
+	{
+		Assumptions.assumeTrue( TestSettings.NEO4J_VERSION.isAtLeastVersion( Neo4jVersion.NEO4J_VERSION_400 ),
+								"This test only applies to 4.0 onwards ssl configuration");
+	}
+
+	private GenericContainer createBasicContainer()
+	{
+		GenericContainer container = new GenericContainer( TestSettings.IMAGE_ID );
+		container.withEnv( "NEO4J_AUTH", "none" )
+				 .withEnv( "NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes" )
+				 .withExposedPorts( 7473, 7474, 7687 )
+				 .withLogConsumer( new Slf4jLogConsumer( log ) )
+				 .waitingFor( Wait.forHttp( "/" )
+								  .forPort( 7474 )
+								  .withStartupTimeout( Duration.ofSeconds( 60 ) ) );
+		return container;
+	}
+
+	@Test
+	void testHttpsNotAlreadyEnabled()
+	{
+		try(GenericContainer container = createBasicContainer())
+		{
+			container.start();
+			Assertions.assertTrue( container.isRunning(), "container did not start" );
+			Assertions.assertFalse( isHttpsResponsive( container ), "HTTPS was enabled when we didn't supply certificates" );
+		}
+	}
+
+	@Test
+	void testCanEncryptHttps() throws Exception
+	{
+		try(GenericContainer container = createBasicContainer())
+		{
+			Path sslFolder = createSSLDirectory( container, "https" );
+			HostFileSystemOperations.mountHostFolderAsVolume( container, sslFolder, "/var/lib/neo4j/certificates" );
```

Comment:
why are we mounting to `/var/lib/neo4j/..` here? can we have a comment to explain why we're doing this rather than mounting to `/ssl`?